applications/FreeSurfaceApplication/python_scripts/edgebased_levelset_solver.py

```diff
@@ -83,10 +79,11 @@ def __init__(self,
         self.compute_porous_resistance_law = EdgeBasedLevelSetSolver.PorousResistanceComputation[self.settings["compute_porous_resistance_law"].GetString()]
 
         # Other linear solvers considered in the original script:
-        # - CGSolver(1e3, 5000)
-        # - CGSolver(1e3, 500, DiagonalPreconditioner())
-        # - BICGSTabSolver(1e3, 5000, DiagonalPreconditioner())
-        self.pressure_linear_solver = KratosMultiphysics.BICGSTABSolver(1e-3, 5000)
+        # - KratosMultiphysics.CGSolver(1e3, 5000)
+        # - KratosMultiphysics.CGSolver(1e3, 500, KratosMultiphysics.DiagonalPreconditioner())
+        # - KratosMultiphysics.BICGSTabSolver(1e3, 5000, KratosMultiphysics.DiagonalPreconditioner())
+        pDiagPrecond = KratosMultiphysics.DiagonalPreconditioner()
+        self.pressure_linear_solver = KratosMultiphysics.BICGSTABSolver(1e-3, 5000, pDiagPrecond)
```

Comment:
Maybe in the future you can consider using the `python_linear_solver_factory` in here. In that way you could easily customize the linear solver settings and (more important IMO) try the AMGCL. Hopefully it will be faster.

Reply:
Done